appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBTimerService.java

```diff
@@ -1552,6 +1552,17 @@ protected void cancelTimerSynchronization(EJBContextImpl context_,
             }
         }
     }
+    
+    /**
+     * Sets failed times to be rescheduled rather than removed.
+     * This is called by {@link fish.payara.ejb.timer.hazelcast.HazelcastTimerStore},
+     * as when an instance is removed from a cluster it is possible that a delivery
+     * could fail in the time in which an instance is removed from the cluster and before
+     * the timer is recreated.
+     */
+    protected void enableRescheduleTimers() {
+        rescheduleFailedTimer = true;
```

Comment:
I really wonder if this field shouldn't be thread-safe. This class has a number of fields I'd think should be thread-safe as I would assume that methods are called by different threads. Is there any reasoning why this would not be the case?

Reply:
This method does not need to be thread-safe as it is only called by the constructor of `HazelcastTimerStore` which will only occur once on startup. The field effectively final, the but I can't make the field final due to the `initProperties()` method also setting this based on the config, which is what `HazelcastTimerStore` overrides to always be true (as that is the point of a clustered timer).

Reply:
I see - technically I agree it doesn't need it. But how should someone know looking at the code unless he also knows what you just explained or finds this out by analysing the code. A short comment would help.

Reply:
There is a comment on the method already